lib/processors/themeBuilderWorker.js

```diff
@@ -0,0 +1,88 @@
+import workerpool from "workerpool";
+import themeBuilder from "./themeBuilder.js";
+import fsInterface from "@ui5/fs/fsInterface";
+import {createAdapter, createWriterCollection, createResource} from "@ui5/fs/resourceFactory";
+
+/**
+ * Task to build library themes.
+ *
+ * @private
+ * @function default
+ * @static
+ *
+ * @param {object} parameters Parameters
+ * @param {object} parameters.allResources Array of Uint8Array transferable objects
+ * 	with all the resources needed to build the css files. By nature those are @ui5/fs/Resource.
+ * @param {object} parameters.themeResources Input array of Uint8Array transferable objects
+ * 	that are the less sources to build upon. By nature those are @ui5/fs/Resource.
+ * @param {object} parameters.options Less compiler options
+ * @returns {Promise<Uint8Array[]>}
+ */
+export default async function execThemeBuild({
+	allResources = [],
+	themeResources = [],
+	options = {}
+}) {
+	const fsThemeResources = themeResources.map((res) => {
+		// res.transferable is an Uint8Array object and needs to be casted
+		// to a Buffer in order to be read correctly.
+		return createResource({path: res.path, buffer: Buffer.from(res.transferable)});
+	});
+	const fsAllResources = allResources.map((res) => {
+		// res.transferable is an Uint8Array object and needs to be casted
+		// to a Buffer in order to be read correctly.
+		return createResource({path: res.path, buffer: Buffer.from(res.transferable)});
```

Comment:
Duplicate code which can easily be avoided

Reply:
Done